Sources/Sentry/SentryClient.m

```diff
@@ -120,18 +120,26 @@ - (NSString *_Nullable)captureException:(NSException *)exception
     SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
     [self attachStacktrace:event];
     event.message = exception.reason;
-    return [self captureEvent:event withScope:scope];
+    return [self sendEvent:event withScope:scope];
 }
 
 - (NSString *_Nullable)captureError:(NSError *)error withScope:(SentryScope *_Nullable)scope
 {
     SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
     [self attachStacktrace:event];
     event.message = error.localizedDescription;
-    return [self captureEvent:event withScope:scope];
+    return [self sendEvent:event withScope:scope];
 }
 
 - (NSString *_Nullable)captureEvent:(SentryEvent *)event withScope:(SentryScope *_Nullable)scope
+{
+    if ([self.options.attachStacktrace boolValue]) {
+        [self attachStacktrace:event];
+    }
+    return [self sendEvent:event withScope:scope];
+}
```

Comment:
M: I think this should be part of `prepareEvent`
every "mutation" of the event should live there.